File: apps/android/app/src/main/java/chat/mural/network/OpenRouterModels.kt
package chat.mural.network

/**
 * Central OpenRouter model routing for Mural (mirrors [shared/openrouter/models.json]).
 * LLM tutoring, STT and TTS each use a dedicated model slug on OpenRouter.
 */
object OpenRouterModels {
    const val BASE_HOST = "openrouter.ai"
    const val LLM = "deepseek/deepseek-v4-pro-0813"
    const val STT = "qwen/qwen3-asr-flash-2026-02-10"
    const val TTS = "hexgrad/kokoro-82m"
    const val DEFAULT_TTS_VOICE = "af_bella"

    /** Cost control: keep spoken replies short so Kokoro bills fewer characters per turn. */
    const val VOICE_BREVITY_PROMPT =
        "Antworte im Sprachmodus stets kurz, direkt und in maximal zwei kurzen Sätzen. Keine Aufzählungen, keine Meta-Kommentare."

    fun ttsVoiceForLocale(locale: String?): String = when (locale?.lowercase()) {
        "ru-ru", "de-de", "es-es", "en-us", "fr-fr", "it-it", "pt-br", "zh-cn", "nb-no" -> DEFAULT_TTS_VOICE
        else -> DEFAULT_TTS_VOICE
    }

    fun isOpenRouterKey(key: String): Boolean {
        val trimmed = key.trim()
        return trimmed.startsWith("sk-or-") && trimmed.length >= 24 && trimmed.none(Char::isWhitespace)
    }
}

File: apps/ios/App/OpenRouterModels.swift
import Foundation

/// Mirrors shared/openrouter/models.json — replace slugs when your housing park IDs are final.
enum OpenRouterModels {
    static let baseURL = URL(string: "https://openrouter.ai/api/v1")!
    static let llm = "deepseek/deepseek-v4-pro-0813"
    static let stt = "qwen/qwen3-asr-flash-2026-02-10"
    static let tts = "hexgrad/kokoro-82m"
    static let defaultVoice = "af_bella"
    static let voiceBrevity = "Antworte im Sprachmodus stets kurz, direkt und in maximal zwei kurzen Sätzen."

    static func isOpenRouterKey(_ key: String) -> Bool {
        let trimmed = key.trimmingCharacters(in: .whitespacesAndNewlines)
        return trimmed.hasPrefix("sk-or-") && trimmed.count >= 24 && !trimmed.contains(where: \.isWhitespace)
    }
}
